sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/EmptyMetricData.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.data;
+
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.resources.Resource;
+
+final class EmptyMetricData implements MetricData {
+
+  static final EmptyMetricData INSTANCE = new EmptyMetricData();
+
+  private EmptyMetricData() {}
+
+  @Override
+  public Resource getResource() {
+    throw new UnsupportedOperationException("EmptyMetricData does not support getResource().");
+  }
```

Comment:
Let's go ahead and add unit tests for these methods. Also in case a caller doesn't call `isEmpty` should we return default values instead of throwing?

Reply:
I thought about returning default values but couldn't come up with reasonable "empty" values for all the fields. And I didn't want to make any of the fields `@Nullable`, or have inconsistent behavior where some accessors throw and others return "empty" values. 

If we do our job right and check for `isEmpty()` in the SDK, this _shouldn't_ ever escape to users.